src/ralph/lib/openstack/client.py

```diff
@@ -193,3 +201,151 @@ def get_keystone_projects(self):
         else:
             projects_resource = self.keystone_client.tenants
         yield from projects_resource.list()
+
+
+class RalphOpenStackInfrastructureClient:
+    def __init__(self, openstack_provider_name):
+        super().__init__()
+        self.DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+        self.openstack_provider_name = openstack_provider_name
+        self.clients = self._get_instances_from_settings()
+
+    def _get_instances_from_settings(self):
+        """
+        Filter instances from OPENSTACK_INSTANCES (from settings) and
+        initialize clients only for the ones matching current
+        openstack provider.
+        """
+        clients = []
+        for os_instance in settings.OPENSTACK_INSTANCES:
+            if os_instance.get(
+                'provider', DEFAULT_OPENSTACK_PROVIDER_NAME
+            ) == self.openstack_provider_name:
+                clients.append(RalphOpenstackClient(os_instance))
+        return clients
+
+    @classmethod
+    def _add_flavor(cls, client, flavor, openstack_flavors):
+        flavor_id = flavor['id']
+        new_flavor = {
+            'name': flavor['name'],
+            'cores': flavor['vcpus'],
+            'memory': flavor['ram'],
+            'disk': flavor['disk'] * 1024,
+            'tag': client.site['tag'],
+        }
+        openstack_flavors[flavor_id] = new_flavor
```

Comment:
I think this method shouldn't modify `openstack_flavors`. It's hard to track what is happening with flavors :)
```
openstack_flavors[flavor['id']] = self._make_flavor(client, flavor)
```